### PR TITLE
Add avatar (default or custom) and name if exists

### PR DESCRIPTION
**Pull request by J2TeamNNL**

### CODE REVIEW

File: src/Reddit/Provider.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ protected function mapUserToObject(array $user)
 
         return (new User())->setRaw($user)->map([
             'id'   => $user['id'], 'nickname' => $user['name'],
-            'name' => null, 'email' => null, 'avatar' => $avatar,
+            'name' => empty($user['subreddit']['title']) ? null : $user['subreddit']['title'], 'email' => null, 'avatar' => strtok($user['icon_img'], '?'),
```
Comment (atymic) [inline src/Reddit/Provider.php:76]: What is the content of `$user['subreddit']['title']`? Doesn't look like name.

Comment (atymic) [inline src/Reddit/Provider.php:76]: We're already including the avatar?

Comment (J2TeamNNL) [reply src/Reddit/Provider.php:76]: Oh sorry about that, I didn't notice

Comment (J2TeamNNL) [reply src/Reddit/Provider.php:76]: image
This is display name that user can custom

Comment (atymic) [reply src/Reddit/Provider.php:76]: Are you sure that that ends up in the `subreddit` key?
That doesn't seem to make much sense.

Comment (J2TeamNNL) [reply src/Reddit/Provider.php:76]: image
I pretty sure it is

File: src/Reddit/Provider.php [hunk 2]
```diff
@@ -70,10 +69,16 @@ protected function mapUserToObject(array $user)
                 $avatar = substr($avatar, 0, $querypos);
             }
         }
+        
+        $name = null;
+        //Check if user have display name
```
Comment (atymic) [inline src/Reddit/Provider.php:74]:
```suggestion
        //Check if user has a display name
```

**Pull request merged**